sbom/formatted_reader.go

```diff
@@ -78,6 +87,61 @@ func (f *FormattedReader) Read(b []byte) (int, error) {
 			}
 		}
 
+		// Makes SPDX SBOM more reproducible, see
+		// https://github.com/paketo-buildpacks/packit/issues/368 for more details.
+		if f.format.ID() == "spdx-2-json" {
+			hash, err := hashstructure.Hash(f.sbom.syft, hashstructure.FormatV2, nil)
+			if err != nil {
+				// not tested
+				return 0, fmt.Errorf("failed to modify SPDX SBOM for reproducibility: %w", err)
+			}
+			hashBytes := make([]byte, binary.MaxVarintLen64)
+			binary.PutUvarint(hashBytes, hash)
+
+			var spdxOutput map[string]interface{}
+
+			err = json.Unmarshal(output, &spdxOutput)
+			if err != nil {
+				return 0, fmt.Errorf("failed to modify SPDX SBOM for reproducibility: %w", err)
+			}
+			for k := range spdxOutput {
```

Comment:
Is there a reason to loop over the keys here rather than just doing something like the following?

```go
if namespace, ok := output["documentNamespace"].(string); ok {
  // do work
}
```